client/my-sites/stats/stats-purchase/stats-purchase-checkout-redirect.tsx

```diff
@@ -56,9 +57,10 @@ const getCheckoutBackUrl = ( {
 	const isFromWPAdmin = from.startsWith( 'jetpack' );
 	const isFromMyJetpack = from === 'jetpack-my-jetpack';
 	const isFromPlansPage = from === 'calypso-plans';
+	const isOdysseyStats = config.isEnabled( 'is_running_in_jetpack_site' );
 
 	// Use full URL even though redirecting on Calypso.
-	if ( ! isFromWPAdmin ) {
+	if ( ! isFromWPAdmin && ! isOdysseyStats ) {
```

Comment:
```suggestion
	if ( ! isOdysseyStats ) {
```
Could we just check `isOdysseyStats` since the start of the query string `from` is basically related to `isOdysseyStats`?

Reply:
Thanks for the feedback. In the future, I think it's a yes, but at the stage Jetpack sites will still be purchasing Stats product in Calypso, and we'll need to keep the compatibility before the purchase page is live in Odyssey.